### DISCUSSION TITLE
Cake version compatible with .NETFramework v4.7

### DISCUSSION THREAD

**Discussion by sirmarv** [cake-build/cake#3658] [Q&A]
What version of Cake is compatible with .NETFramework v4.7?

I am trying to install Cake to my project via Nuget Packages on my VS2019.

Please help, turns out Cake is at the base of our architecture and is the only way of creating builds on this particular project.
I am getting the error below, after trying several versions by the way:

`Could not install package 'Cake <version>'. You are trying to install this package into a project that targets '.NETFramework,Version=v4.7', but the package does not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author.				
`

**Comment by nils-a** [comment]
@sirmarv can you explain a bit what you are trying to do? 
Are you trying to reference https://www.nuget.org/packages/Cake/ as a package in a project?

**Reply by sirmarv** [reply]
@nils-a yes i am. See below image:
![cakeerror](https://user-images.githubusercontent.com/10893996/139843065-592afc3d-5cd0-472a-b712-e7b2faf91b1b.PNG)

**Reply by nils-a** [reply]
I'm not sure this will work somehow - maybe someone else can answer that - but to reference a package as a library that package needs to be published as a library. Cake is published as a "tool" or "utility".

Are you looking into how to build your Cake based project, or are you trying to extend on Cake features?
